### PR TITLE
OpenSHMEM v1.5rc2

### PR DESCRIPTION
**Pull request by jdinan**
The OpenSHMEM specification committee is pleased to announce the availability of the second OpenSHMEM 1.5 release candidate. This draft contains the contributions from the section committees as indicated below.

We encourage members of the specification committee and the broader community to review this draft in advance of the [upcoming release candidate meeting on April 27](https://github.com/openshmem-org/specification/wiki/April-27th). Please log any feedback as github issues and assign to the OpenSHMEM 1.5 milestone and RC2 project to ensure they are addressed by the committee.

Thank you to everyone who has contributed to this draft of the OpenSHMEM specification!

OpenSHMEM Draft Contains:
* RaymondMichael/sec/frontmatter - 7ff6bd3e [diff](https://github.com/openshmem-org/specification/pull/415/commits/7bcf97d9c18332b9df416a6717fea64a6d64d038)
* manjugv/section/library_setup - f4db7e37 No changes since RC1
* jdinan/sec/ctx-teams - 41982f83 [diff](https://github.com/openshmem-org/specification/pull/415/commits/489945eb120cef34701e12c894e9bce30786e9db)
* shamisp/sec/rma/1.5.0 - cf60bcc6 [diff](https://github.com/openshmem-org/specification/pull/415/commits/31ee331721361805100c7b440d9145c90c5e3662)
* nspark/section/collectives - e21f4c55 [diff](https://github.com/openshmem-org/specification/pull/415/commits/4fad78eda22617b14e1d14efe7ed480ca496f35b)
* naveen-rn/naveen-rn/openshmem-1.5 - 25eef87e [diff](https://github.com/openshmem-org/specification/pull/415/commits/917d740654fa8fc3da2b2773d9d06492b4ffb541)
* wrrobin/sec/profiling-interface - 14a0a12b [diff](https://github.com/openshmem-org/specification/pull/415/commits/3a6b0b82b341f26d9bf72d88b7e42e6425db0f28)
* BryantLam/sec/backmatter - 384d1642 [diff](https://github.com/openshmem-org/specification/pull/415/commits/532c09398917b3fe87b323da3c8cba6efb148eff)

Attachment: [openshmem-1.5rc2.pdf](https://github.com/openshmem-org/specification/files/4439982/openshmem-1.5rc2.pdf)

### CODE REVIEW

File: content/shmem_ctx_get_team.tex [hunk 1]
```diff
@@ -23,30 +23,17 @@
...
-  \FUNC{shmem\_ctx\_create}, the \VAR{team} is assigned the handle value
+  \FUNC{shmem\_ctx\_create}, \VAR{team} is assigned the handle value
   \LibHandleRef{SHMEM\_TEAM\_WORLD}.
 
   If \VAR{ctx} compares equal to \LibConstRef{SHMEM\_CTX\_INVALID},
   then \VAR{team} is assigned the value
   \LibConstRef{SHMEM\_TEAM\_INVALID} and a nonzero value is returned.
   If \VAR{ctx} is otherwise invalid, the behavior is undefined.
-
-  If \VAR{team} is a null pointer, the behavior is undefined.
```
Comment (jdinan) [inline content/shmem_ctx_get_team.tex:34]: Put this line back.

File: content/teams_intro.tex [hunk 2]
```diff
@@ -1,21 +1,21 @@
 The \acp{PE} in an \openshmem program communicate using either
-point-to-point routines---such as \ac{RMA} and \ac{AMO} routines---which specify the \ac{PE} number of the target
-\ac{PE}, or collective routines, which operate over a set of \acp{PE}.
-In \openshmem, teams allow programs to group a set of \acp{PE} for
+point-to-point routines --- such as \ac{RMA} and \ac{AMO} routines --- that specify the \ac{PE} number of the target
```
Comment (jdinan) [inline content/teams_intro.tex:2]: @nspark @jdinan Resolve spaces around `---`.

File: content/teams_intro.tex [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
 
 An \openshmem team may be predefined (i.e., provided by the \openshmem
-library) or defined by the \openshmem program.
-A program-defined team is created by ``splitting'' a parent team into
-one or more new teams---each with some subset of \acp{PE} of the
-parent team---via one of the \FUNC{shmem\_team\_split\_*} routines.
+library) or defined by the \openshmem application.
+An application-defined team is created by ``splitting'' a parent team into
+one or more new teams --- each with some subset of \acp{PE} of the
+parent team --- via one of the \FUNC{shmem\_team\_split\_*} routines.
```
Comment (jdinan) [inline content/teams_intro.tex:18]: Resolve `---`.

File: content/teams_intro.tex [hunk 4]
```diff
@@ -50,26 +50,28 @@ \subsubsection*{Thread Safety}
...
+teams may be performed in parallel.
 
 \subsubsection*{Collective Ordering}
 
 In \openshmem, a team object encapsulates resources used to communicate
 between \acp{PE} in collective operations. When calling multiple subsequent
-collective operations on a team, the collective operations---along with any
-relevant team based resources---are matched across the \acp{PE} in the team
+collective operations on a team, the collective operations --- along with any
+relevant team based resources --- are matched across the \acp{PE} in the team
```
Comment (jdinan) [inline content/teams_intro.tex:62]: Resolve `---`.

File: main_spec.tex [hunk 5]
```diff
@@ -435,10 +435,34 @@ \subsubsection{\textbf{SHMEM\_SIGNAL\_WAIT\_UNTIL}}\label{subsec:shmem_signal_wa
...
+  Blocking \OPR{Get}                 &   &   \\
+  Blocking \OPR{AMO}                 & X & X \\
+  Blocking \OPR{put-with-signal}     & X & X \\
+  Non-blocking \OPR{Put}             & X & X \\
+  Non-blocking \OPR{Get}             &   & X \\
+  Non-blocking \OPR{AMO}             & X\footnote{\openshmem fence routines does
+  not guarantee order of delivery of values fetched by nonblocking \ac{AMO}
+  routines.}
+                                         & X \\
+  Non-blocking \OPR{put-with-signal} & X & X \\ \hline
```
Comment (jdinan) [inline main_spec.tex:459]: Remove "-" in nonblocking.

**Comment by wrrobin** [comment]
There is a large white space gap on page 5 of the pdf. Seems like it needs to be fixed.

File: content/synchronization_model.tex [hunk 6]
```diff
@@ -108,3 +87,25 @@
 \hline
 \end{tabular}
 \clearpage
+
+\begin{tabular}{p{0.2\textwidth} | p{0.7\textwidth}}
+Collective synchronization over \\
+\FUNC{shmem\_team\_sync}
+&
+\raisebox{-\totalheight}{\includegraphics[width=0.8\textwidth]{figures/sync}}
```
Comment (jdinan) [inline content/synchronization_model.tex:95]: Figures are spilling into the margin. Can we try `width=\linewidth`?

Comment (jdinan) [reply content/synchronization_model.tex:95]: May need to remove one of the PEs from the figure in order to make the figure still readable.

File: content/programming_model_overview.tex [hunk 7]
```diff
@@ -99,15 +99,15 @@
...
   \item \OPR{Quiet}: The \ac{PE} calling quiet ensures remote completion of remote access
       operations and stores to symmetric data objects.
   \item \OPR{Barrier}: All or some \acp{PE} collectively synchronize and ensure
       completion of all remote and local updates prior to any \ac{PE} returning
       from the call.
-  \item \OPR{Wait and Test}: A PE calling a point-to-point synchronization
+  \item \OPR{Wait and Test}: A \ac{PE} calling a point-to-point synchronization
       routine ensures the value of a local symmetric object meets a specified
       condition.  Wait operations block until the specified condition is
       met, whereas test operations return immediately and indicate whether or
```
Comment (jdinan) [inline content/programming_model_overview.tex:113]: Add put with signal and teams to prog model overview.

**Comment by jdinan** [comment]
Changes approved at April 27, 2020 meeting. To be ratified at June meeting.

**Comment by jdinan** [comment]
@wrrobin Whitespace fixed via doc edits. Thanks!

File: content/programming_model_overview.tex (hunk 7 above)
Comment (jdinan) [reply content/programming_model_overview.tex:113]: Captured in #430

**Pull request merged**